operator/pkg/metrics/server.go

```diff
@@ -0,0 +1,211 @@
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+	"unicode"
+
+	"github.com/benbjohnson/clock"
+	"github.com/gorilla/mux"
+	metricsv1alpha1 "github.com/keptn/lifecycle-toolkit/operator/apis/metrics/v1alpha1"
+	"github.com/open-feature/go-sdk/pkg/openfeature"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+type Metrics struct {
+	gauges map[string]prometheus.Gauge
+}
+
+var metrics Metrics
+
+var instance *serverManager
+var smOnce sync.Once
+
+type serverManager struct {
+	server        *http.Server
+	ticker        *clock.Ticker
+	ofClient      *openfeature.Client
+	exposeMetrics bool
+}
+
+func StartServerManager(ctx context.Context, exposeMetrics bool) {
```

Comment:
can we add some unit tests for this component?

Reply:
Probably yes, but in that case we would need to introduce a mock struct for serverManager and export the methods this struct, so that goes basically against the comment you provided before. Or what kind of unit testing do you have in mind?

Reply:
not sure if I understand the question, but why would we need to mock the serverManager?